edb/edgeql-chumsky/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+mod ast;
+mod inheritance;
+mod parser;
+
+use std::ops::Range;
+
+use chumsky::Parser;
+use edgeql_parser::{position::Pos, tokenizer::TokenStream};
+use pyo3::types::PyModule;
+use pyo3::*;
+
+use crate::inheritance::{IntoPyChild, IntoPyObject};
+use crate::parser::expr_stmt;
+
+#[pymodule]
+fn pyo3_example(_py: Python<'_>, m: &PyModule) -> PyResult<()> {
+    m.add_function(wrap_pyfunction!(parse, m)?)?;
+    m.add_class::<crate::ast::ExprPyAttrs>()?;
+    m.add_class::<crate::ast::QueryPyAttrs>()?;
+    m.add_class::<crate::ast::PipelinedQueryPyAttrs>()?;
+    m.add_class::<crate::ast::SelectQueryPyAttrs>()?;
+    Ok(())
+}
+
+#[pyfunction]
+fn parse(text: String) -> PyResult<PyObject> {
+    let mut token_stream = TokenStream::new(&text);
+    let mut tokens = Vec::new();
```

Comment:
Eventually have to get rid of `TokenStream` from `combine`. Initially I wanted to use `combine` from parser instead of chumsky. But that was long time ago. Probably `TokenStream` should emit something more chumsky-compatible and not buffer all the tokens into a vector.